### PR TITLE
GH-11: Add generic `is` type guard

### PR DESCRIPTION
**Pull request by JoakimDeak**

### CODE REVIEW

**Comment by fabian-hiller** [comment]
Thank you for creating this PR. I'll check your code later.

**Review by jasikpark — APPROVED** [review]
looks great to me 💜 thx for building this, using it now while i wait for the merge

**Comment by fabian-hiller** [comment]
We are still considering whether there are alternative names to `is` in #11. If there is no better alternative, I think we stick with `is` and then I check the implementation.

File: library/src/methods/is/is.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+ *
+ * @param input The input to be tested
+ *
+ * @returns A type predicate for the given schema
+ */
+export function is<TSchema extends BaseSchema>(
+  schema: TSchema,
+  input: unknown
+): input is Output<TSchema> {
+  return safeParse(schema, input).success;
```
Comment (fabian-hiller) [inline library/src/methods/is/is.ts:17]: `safeParse` has more functionality and thus code than `is` technically required. Therefore we could change this line to the following code and remove the import above.

```ts
try {
  schema.parse(input);
  return true;
} catch (error) {
  return false;
}
```

File: library/src/methods/is/is.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import { BaseSchema, Output } from '../../types';
+import { safeParse } from '../safeParse';
+
+/**
+ * Type guard for a given schema
+ *
+ * @param schema The scheme to be used
+ *
```
Comment (fabian-hiller) [inline library/src/methods/is/is.ts:8]: Can you remove this empty line?

**Comment by fabian-hiller** [comment]
If we add `is`, we should also add `isAsync`

File: library/src/methods/is/is.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+ * @param input The input to be tested
+ *
+ * @returns A type predicate for the given schema
+ */
+export function is<TSchema extends BaseSchema>(
+  schema: TSchema,
+  input: unknown
+): input is Output<TSchema> {
```
Comment (fabian-hiller) [inline library/src/methods/is/is.ts:16]: Using `Output` here is an error, because transformations have no effect. Therefore `Input` must be used.

**Comment by fabian-hiller** [comment]
I'll revise your code in a moment and then merge it.

**Comment by fabian-hiller** [comment]
> If we add `is`, we should also add `isAsync`

This does not seem to be technically possible at the moment: https://github.com/microsoft/TypeScript/issues/37681

**Pull request merged**